### PR TITLE
Wrong usage of psql

### PR DESCRIPTION
**Pull request by jf-viguier**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Wrong usage of psql in an IN sql request
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | N/A
| How to test?  | Code refacto

### CODE REVIEW

**Review by matks — COMMENTED** [review]
Sounds good but QA team will be tasked to be sure this change does not introduce any unwanted side-effects. Could you explain them what user scenario should be followed to verify `getCategories()` change is valid?

File: ps_categorytree.php [hunk 1]
```diff
@@ -161,7 +161,7 @@ private function getCategories($category)
 			AND c.id_category IN (
 				SELECT id_category
 				FROM `' . _DB_PREFIX_ . 'category_group`
-				WHERE `id_group` IN (' . pSQL(implode(', ', Customer::getGroupsStatic((int) $this->context->customer->id))) . ')
+				WHERE `id_group` IN (' . implode(', ', array_map('intval', Customer::getGroupsStatic((int) $this->context->customer->id))) . ')
```
Comment (atomiix) [inline ps_categorytree.php:164]: It seems that the whole usage of pSQL is useless here, this is not a value took from a user, and [it's already casted to `int` in `Customer::getGroupsStatic()`](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/classes/Customer.php#L1071)

Comment (jf-viguier) [reply ps_categorytree.php]: Thanks @atomiix I've deleted the psql

**Comment by jf-viguier** [comment]
@matks it should be tested by a developer, no side effect possible. It's just a security concern, a wrong usage of psql

**Review by atomiix — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
Thanks @jf-viguier

**Pull request merged**